conans/client/cmd/export.py

```diff
@@ -179,43 +190,51 @@ def cmd_export(app, conanfile_path, name, version, user, channel, keep_source,
     return ref
 
 
-def _capture_export_scm_data(conanfile, conanfile_dir, destination_folder, output, scm_src_file):
-
-    if os.path.exists(scm_src_file):
-        os.unlink(scm_src_file)
-
+def _capture_scm_auto_fields(conanfile, conanfile_dir, destination_folder, output, ignore_dirty):
+    """Deduce the values for the scm auto fields or functions assigned to 'url' or 'revision'
+       and replace the conanfile.py contents.
+       Returns a tuple with (scm_data, path_to_scm_local_directory)"""
     scm_data = get_scm_data(conanfile)
     if not scm_data:
-        return
+        return None, None
 
     # Resolve SCMData in the user workspace (someone may want to access CVS or import some py)
     scm = SCM(scm_data, conanfile_dir, output)
     captured = scm_data.capture_origin or scm_data.capture_revision
 
-    if scm_data.url == "auto":
-        origin = scm.get_qualified_remote_url(remove_credentials=True)
-        if not origin:
-            raise ConanException("Repo origin cannot be deduced")
-        if scm.is_local_repository():
-            output.warn("Repo origin looks like a local path: %s" % origin)
-        output.success("Repo origin deduced by 'auto': %s" % origin)
-        scm_data.url = origin
-
-    if scm_data.revision == "auto":
-        if not scm.is_pristine():
-            output.warn("Repo status is not pristine: there might be modified files")
-        scm_data.revision = scm.get_revision()
-        output.success("Revision deduced by 'auto': %s" % scm_data.revision)
+    if not captured:
+        # We replace not only "auto" values, also evaluated functions (e.g from a python_require)
+        _replace_scm_data_in_conanfile(os.path.join(destination_folder, "conanfile.py"), scm_data)
+        return scm_data, None
+
+    if not scm.is_pristine() and not ignore_dirty:
+        output.warn("There are uncommitted changes, skipping the replacement of 'scm.url' and "
+                    "'scm.revision' auto fields. Use --ignore-dirty to force it. The 'conan "
+                    "upload' command will prevent uploading recipes with 'auto' values in these "
+                    "fields.")
+        local_src_path = scm.get_local_path_to_url(scm_data.url) \
+            if scm_data.url != "auto" else conanfile_dir
+        return scm_data, local_src_path
+    else:
```

Comment:
This ``else`` can be removed, and the structure of this function will be ``return`` based, like in the above ``if not captured``